src/openrct2/EditorObjectSelectionSession.cpp

```diff
@@ -483,44 +486,57 @@ int32_t window_editor_object_selection_select_object(uint8_t bh, int32_t flags,
 
         if (maxObjects <= _numSelectedObjectsForType[objectType])
         {
-            set_object_selection_error(bh, STR_OBJECT_SELECTION_ERR_TOO_MANY_OF_TYPE_SELECTED);
-            return 0;
+            set_object_selection_error(isMasterObject, STR_OBJECT_SELECTION_ERR_TOO_MANY_OF_TYPE_SELECTED);
+            return false;
         }
 
-        if (objectType == OBJECT_TYPE_SCENERY_GROUP && (flags & INPUT_FLAG_EDITOR_OBJECT_2))
+        if (objectType == OBJECT_TYPE_SCENERY_GROUP && (flags & INPUT_FLAG_EDITOR_OBJECT_SELECT_OBJECTS_IN_SCENERY_GROUP))
         {
             for (const auto& sgEntry : item->SceneryGroupInfo.Entries)
             {
-                if (!window_editor_object_selection_select_object(++bh, flags, &sgEntry))
+                if (!window_editor_object_selection_select_object(++isMasterObject, flags, &sgEntry))
                 {
                     _maxObjectsWasHit = true;
                 }
             }
         }
 
-        if (bh != 0 && !(flags & INPUT_FLAG_EDITOR_OBJECT_1))
+        if (isMasterObject != 0 && !(flags & INPUT_FLAG_EDITOR_OBJECT_1))
         {
             char objectName[64];
             object_create_identifier_name(objectName, 64, &item->ObjectEntry);
             auto ft = Formatter::Common();
             ft.Add<const char*>(objectName);
-            set_object_selection_error(bh, STR_OBJECT_SELECTION_ERR_SHOULD_SELECT_X_FIRST);
-            return 0;
+            set_object_selection_error(isMasterObject, STR_OBJECT_SELECTION_ERR_SHOULD_SELECT_X_FIRST);
+            return false;
         }
 
         if (maxObjects <= _numSelectedObjectsForType[objectType])
         {
-            set_object_selection_error(bh, STR_OBJECT_SELECTION_ERR_TOO_MANY_OF_TYPE_SELECTED);
-            return 0;
+            set_object_selection_error(isMasterObject, STR_OBJECT_SELECTION_ERR_TOO_MANY_OF_TYPE_SELECTED);
+            return false;
         }
 
         _numSelectedObjectsForType[objectType]++;
 
         *selectionFlags |= OBJECT_SELECTION_FLAG_SELECTED;
-        return 1;
+        return true;
     }
 }
 
+bool window_editor_object_selection_select_object(uint8_t isMasterObject, int32_t flags, std::string_view identifier)
```

Comment:
Should this be title case?

Reply:
Perhaps better taken as part of a window cleanup. (Would especially be nice to make it a method if we were to use subclassing.)